xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -111,7 +112,14 @@ private String toStringButEmptyIfNull(Object object)
     {
         List<ListItem> list = getCachedDBList(context);
         if (list == null) {
-            String hqlQuery = getQuery(context);
+            String hqlQuery = null;
+            try {
+                hqlQuery = getQuery(context);
+            } catch (XWikiException e) {
+                LOGGER.error("Failed to get the list", e);
+                list = new ArrayList<ListItem>();
+                return list;
```

Comment:
Doesn't ``hqlQuery`` remain ``null`` if ``getQuery`` fails? And if ``hqlQuery`` remains ``null`` isn't the ``list`` assigned to an empty list just two lines below?